### PR TITLE
Fix domain registration error handling in cassandra

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
Fix domain registration error handling in cassandra

**Why?**
DomainAlreadyExists error may be returned even if the domain is not actually registered when there's concurrent domain registration requests due to CAS failure on domain metadata.

**How did you test it?**
Persistence unit test

**Potential risks**

### CODE REVIEW

File: common/persistence/nosql/nosqlplugin/cassandra/domain.go [hunk 1]
```diff
@@ -224,13 +224,31 @@ func (db *cdb) InsertDomain(
...
-			msg := fmt.Sprintf("Domain already exists.  DomainId: %v", domain)
-			db.logger.Warn(msg)
-			return errConditionFailed
+		for {
+			// first iter MapScan is done inside MapExecuteBatchCAS
+			domain := previous["name"].(string)
+			if domain == row.Info.Name {
+				db.logger.Warn("Domain already exists", tag.WorkflowDomainName(domain))
+				return &types.DomainAlreadyExistsError{
+					Message: fmt.Sprintf("Domain %v already exists", previous["domain"]),
```
Comment (yux0) [inline common/persistence/nosql/nosqlplugin/cassandra/domain.go:233]: nit: domain

**Review by yux0 — APPROVED** [review]

File: common/persistence/nosql/nosqlplugin/cassandra/domain.go (hunk 1 above)
Comment (yycptt) [reply common/persistence/nosql/nosqlplugin/cassandra/domain.go]: Updated.

File: common/persistence/persistence-tests/metadataPersistenceV2Test.go [hunk 2]
```diff
@@ -195,6 +195,7 @@ func (m *MetadataPersistenceSuiteV2) TestCreateDomain() {
 		0,
 	)
 	m.Error(err2)
+	fmt.Println("ycyang: ", err2)
```
Comment (mkolodezny) [inline common/persistence/persistence-tests/metadataPersistenceV2Test.go:198]: remove? :)

File: common/persistence/cassandra/cassandraMetadataPersistenceV2.go [hunk 3]
```diff
@@ -90,10 +90,8 @@ func (m *nosqlDomainManager) CreateDomain(
 	err = m.db.InsertDomain(ctx, row)
 
 	if err != nil {
-		if m.db.IsConditionFailedError(err) {
```
Comment (mkolodezny) [inline common/persistence/cassandra/cassandraMetadataPersistenceV2.go:93]: did we reproduce this line on prod?

File: common/persistence/persistence-tests/metadataPersistenceV2Test.go (hunk 2 above)
Comment (yycptt) [reply common/persistence/persistence-tests/metadataPersistenceV2Test.go]: 👍 Thanks for catching! Removed.

File: common/persistence/cassandra/cassandraMetadataPersistenceV2.go (hunk 3 above)
Comment (yycptt) [reply common/persistence/cassandra/cassandraMetadataPersistenceV2.go:93]: We don't have concurrent domain registration on prod, so won't be able to repro there. I can easily repro this on my local though, for example starting canary with two non-exists domains in the config.

**Review by yux0 — APPROVED** [review]

**Pull request merged**